src/activity-logger/background/index.js

```diff
@@ -9,3 +9,68 @@ browser.webNavigation.onCommitted.addListener(({url, tabId, frameId}) => {
     maybeLogPageVisit({url, tabId})
 
 })
+
+
+//********************************************************************************/
+var timestamp = new Date().getTime;
+
+var number_of_opened_tabs = 0;
+
+var tabList = [];
+
+
+/*chrome.runtime.onMessage.addListener(function(response,sender,sendResponse){
+    alert(response);
+});*/
+
+
+browser.tabs.onCreated.addListener(function (tab){
+    number_of_opened_tabs = number_of_opened_tabs + 1;
+    //console.log('total opened tabs: ' + number_of_opened_tabs);
+    var t = {
+        id : tab.id,
+        activetime : 0,
+        isActive : false,
+        lastActivatedtime : Number(new Date().getTime())
+    };
+    tabList.push(t);
+    console.log('a new Tab created : ' + tab.id);
+})
+
+browser.tabs.onRemoved.addListener(function (id){
+    number_of_opened_tabs = number_of_opened_tabs - 1;
+    //console.log('total opened tabs: ' + number_of_opened_tabs);
+    for (var index in tabList){
+        if(tabList[index].id == id){
```

Comment:
do you expect more than one match? if not, you should probably use array.find instead.
Also, is there any reason to use an array instead of a hashmap using the tab.id as key?